var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -161,8 +161,8 @@ def setup_build_environment(self, env):
             env.set('FFLAGS', '-fallow-argument-mismatch')
 
     def setup_dependent_build_environment(self, env, dependent_spec):
-        # On Cray, the regular compiler wrappers *are* the MPI wrappers.
-        if 'platform=cray' in self.spec:
+        # For Cray MPIs, the regular compiler wrappers *are* the MPI wrappers.
```

Comment:
IMO it would be worth explicitly stating that cray-provided MPIs have "cray" in the module name (and possibly provide an example name).